### PR TITLE
NETOBSERV-1244: Added flow type transformation rule in FLP configuration

### PR DESCRIPTION
**Pull request by OlivierCazade**
## Description

Added flow type transformation rule in FLP configuration

## Dependencies

Related FLP PR:
https://github.com/netobserv/flowlogs-pipeline/pull/554

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [X] Does this PR require a product release notes entry?
  * [X] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [X] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: controllers/flp/flp_pipeline_builder.go
Comment (OlivierCazade) [inline controllers/flp/flp_pipeline_builder.go:76]: TODO once FLP PR is merged: update FLP dependency and change this string using FLP API

**Comment by OlivierCazade** [comment]
/hold
We need to merge FLP PR first and then update this one

File: api/v1beta1/flowcollector_types.go
Comment (jpinsonneau) [inline api/v1beta1/flowcollector_types.go:667]: :+1:  love it ! It will simplify the display a lot !

File: controllers/consoleplugin/config/static-frontend-config.yaml
Comment (jpinsonneau) [inline controllers/consoleplugin/config/static-frontend-config.yaml:714]: To use `autocomplete` component you will need to implement `getOptions` on plugin side:
https://github.com/netobserv/network-observability-console-plugin/blob/be4ea4532962056f32080cd29bc99ce1fe77ca89/web/src/utils/filter-definitions.ts#L253-L297

We could even improve that to something more generic for `Namespaces`, `Names`, `Kinds` and this new `FlowType` since all of these rely on [autoCompleteCache](https://github.com/netobserv/network-observability-console-plugin/blob/eadc9beff97a2deef33d0494fc080ed3e31e7612/web/src/utils/autocomplete-cache.ts).

That would require a new route, sending field name as argument and refactor [resources.go](https://github.com/netobserv/network-observability-console-plugin/blob/0c2cebc3a144b9e3c24ad266f9af826e5b5aedad/pkg/handler/resources.go) accordingly. WDYT ?

Comment (OlivierCazade) [reply controllers/consoleplugin/config/static-frontend-config.yaml]: I changed to text. I think this is enough since the placeholder already hint the user that this is only infra or app.

And I agree that as a follow up to the generic column, we could have a follow up task to have generic auto complete for generic column.

**Comment by OlivierCazade** [comment]
Updated operator to use this PR:
https://github.com/netobserv/flowlogs-pipeline/pull/554

Until it is merged, to build this branch it is required to use go replace to point at the FLP branch.

**Review by jpinsonneau — APPROVED** [review]
Code looks good to me, thanks @OlivierCazade !

**Comment by jotak** [comment]
I'm also glad to see these awful infra/app filters looking so much better now :-)

One thing that is not a blocker but I think I can do that quickly as a follow-up, if you all agree, is to find another name than "flow type" ... there's so many ways to interpret "type", it's too overloaded and could mean totally different things in different context. I think "layer" is quite common to refer to things like infrastructure / applications, wdyt?

**Review by jotak — APPROVED** [review]

**Comment by jpinsonneau** [comment]
Changes looks good, flow layer will be better indeed :smiley_cat:

**Review by jotak — APPROVED** [review]

**Comment by jotak** [comment]
Thinking about the upgrade process, users will still have their old quick filters. We might have to add in the release note that they could do the change manually in their config (although the old filters will continue to work)

**Comment by OlivierCazade** [comment]
As discussed in scrum today, this will be tested post merge.

**Pull request merged**
